cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -253,11 +278,13 @@ void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
     }
   }
 
+  _stream.synchronize();
```

Comment:
I don't think this sync is necessary.  The `error_code.value()` call will sync all the D2H copies above, and the H2D copies can wait for the sync at the end of the function.

Reply:
Hmm.  The error_code.value() doing the sync is a little bit too subtle for me.   Your average reader isn't going to recognize this easily.  I think the right thing to do here is to just move this particular synchronize right after the error_code.value(). 

Even though it won't technically be doing anything, I like the explicitness.